modules/core/04-channel/client/cli/tx.go

```diff
@@ -0,0 +1,160 @@
+
+import (
+	"fmt"
+	"github.com/spf13/cobra"
+	"regexp"
+	"slices"
+	"strings"
+
+	"github.com/cosmos/cosmos-sdk/client"
+	"github.com/cosmos/cosmos-sdk/client/flags"
+	"github.com/cosmos/cosmos-sdk/client/tx"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/cosmos-sdk/version"
+	govtypesv1 "github.com/cosmos/cosmos-sdk/x/gov/types/v1"
+
+	"github.com/cosmos/ibc-go/v8/modules/core/04-channel/types"
+	ibcexported "github.com/cosmos/ibc-go/v8/modules/core/exported"
+)
+
+const (
+	flagMetadata    = "metadata"
+	flagSummary     = "summary"
+	flagTitle       = "title"
+	flagJSON        = "json"
+	flagPortPattern = "port-pattern"
+	flagExpedited   = "expedited"
+	flagChannelIDs  = "channel-ids"
+)
+
+func newUpgradeChannelsCmd() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "upgrade-channels",
+		Short: "Upgrade IBC channels",
+		Long: `Submit a governance proposal to upgrade all open channels whose port matches a specified pattern 
+(the default is transfer), optionally, specific an exact list of channel IDs with a comma separated list.`,
+		Args:    cobra.ExactArgs(2),
+		Example: fmt.Sprintf(`%s tx %s %s upgrade-channels 10stake "{\"fee_version\":\"ics29-1\",\"app_version\":\"ics20-1\"}"`, version.AppName, ibcexported.ModuleName, types.SubModuleName),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			depositStr := args[0]
+			versionStr := args[1]
+
+			metadata, err := cmd.Flags().GetString(flagMetadata)
+			if err != nil {
+				return err
+			}
+
+			summary, err := cmd.Flags().GetString(flagSummary)
+			if err != nil {
+				return err
+			}
+
+			title, err := cmd.Flags().GetString(flagTitle)
+			if err != nil {
+				return err
+			}
+
+			portPattern, err := cmd.Flags().GetString(flagPortPattern)
+			if err != nil {
+				return err
+			}
+
+			commaSeparatedChannelIDs, err := cmd.Flags().GetString(flagChannelIDs)
+			if err != nil {
+				return err
+			}
+
+			displayJSON, err := cmd.Flags().GetBool(flagJSON)
+			if err != nil {
+				return err
+			}
+
+			expidited, err := cmd.Flags().GetBool(flagExpedited)
+			if err != nil {
+				return err
+			}
+
+			clientCtx, err := client.GetClientQueryContext(cmd)
+			if err != nil {
+				return err
+			}
+
+			queryClient := types.NewQueryClient(clientCtx)
+
+			resp, err := queryClient.Channels(cmd.Context(), &types.QueryChannelsRequest{})
+			if err != nil {
+				return err
+			}
+
+			channelIDs := strings.Split(commaSeparatedChannelIDs, ",")
+
+			var msgs []sdk.Msg
+			pattern := regexp.MustCompile(portPattern)
+
+			for _, ch := range resp.Channels {
+
+				if !channelShouldBeUpgraded(*ch, pattern, channelIDs) {
+					continue
+				}
+
+				// construct a MsgChannelUpgradeInit which will upgrade the specified channel to a specific version.
+				msgUpgradeInit := types.NewMsgChannelUpgradeInit(ch.PortId, ch.ChannelId, types.NewUpgradeFields(ch.Ordering, ch.ConnectionHops, versionStr), clientCtx.GetFromAddress().String())
+				msgs = append(msgs, msgUpgradeInit)
```

Comment:
I left ordering and connection hops unchanged, and only the version string changes. Technically we can allow these fields to be updated, but I think the version string will be all that we need 99% of the time.